File: DmD.WebUi/src/app/models/campaign.ts
import { Map } from "./index";

export class Campaign {
    Id: number;
    Name: string;
    Owner: string;
    CurrentMapId: number;
    Map: Map;

    constructor(campaign: Campaign) {
        this.Id = campaign.Id;
        this.Name = campaign.Name;
        this.Owner = campaign.Owner;
        this.CurrentMapId = campaign.CurrentMapId;
        this.Map = campaign.Map;
    }
}

File: DmD.WebApi/Models/Campaign.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DmD.WebApi.Models
{
    public class Campaign
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Owner { get; set; }

        public int CurrentMapId { get; set; }

        [ForeignKey(nameof(CurrentMapId))]
        public Map Map { get; set; }
    }
}